go-controller/pkg/ovn/base_network_controller_kubevirt.go

```diff
@@ -0,0 +1,52 @@
+package ovn
+
+import (
+	"fmt"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kubevirt"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/util/retry"
+)
+
+// getSwitchNames at some kubevirt scenarios the switch owner the IP is
+// different from the one running the pod
+func (bnc *BaseNetworkController) getSwitchNames(pod *corev1.Pod) (string, string, error) {
+	switchName, err := bnc.getExpectedSwitchName(pod)
+	if err != nil {
+		return "", "", err
+	}
+	switchNameAnnotation, ok := pod.Labels[kubevirt.SubnetSwitchNameLabel]
```

Comment:
nit: I would name this variable `originalSwitchName`.

Reply:
Using a struct now.